### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [andrewjihoonc/snap_that#1]
Hi @andrewjihoonc seems like I could not see where to create new message :(. Can you please update your assignment ?

**Comment by andrewjihoonc** [comment]
Yeah, mine is a little weird. It's more of a chat system depending on the conversation you're in to send a new message because it's a lot more fluid between the users. Anyways, I'll add the feature as soon as possible when I complete this testing assignment. Planning to go back and make it better–it's still at it's early age :P

**Comment by coderschoolreview** [comment]
Hi @dpqhuy, nice work with your assignment 2 :+1:

The goals of this homework are: 
*- to know how to implement authentication from scratch in Rails with the help of the `has_secure_password` class method and the brypt gem (while understanding what they do under the hood) 
*- to design a more complex schema for a web app and be more familiar with ActiveRecord API such as custom foreign keys for `belongs_to/has_many`, using `has_many :through` and building more advanced queries by chaining where, order, and merge 
*- to build more complex forms in views that involve multiple models.

## Quick review:
I see has_secure_password in your User model and you define current_user. Great job! 
Correctly handle user signup in Users controller and user login in Sessions controller
 Nicely define belongs_to/has_many with custom classname and `hasmany :through` for friendship
 I see validations on your models, that's great !.
 Nice using scope to query

## Area to Improve:
You should put some foreign_key in your relationships.
You should pay more attention to the UI
You should test your assignment before putting it on Heroku. I still cannot logout =(



Congrats again on the submission. Everyone has limited time; it's good to see you .... Let us know if you have any feedback.
